src/parsers/cst.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import pkg from 'cst/package.json';
+import loadAndExecute from './utils/loadAndExecute';
+import * as LocalStorage from '../LocalStorage';
+import SettingsRenderer from './utils/SettingsRenderer';
+
+const ID = 'cst';
+
+const options = {
+  strictMode: true
+};
+
+export default {
+  id: ID,
+  displayName: ID,
+  version: pkg.version,
+  homepage: pkg.homepage,
+
+  parse(code) {
+    return loadAndExecute(
+      ['cst'],
+      cst => {
+        let parser = new cst.Parser();
+        if (!options.strictMode) {
+          parser.disableStrictMode();
+        }
+        return parser.parse(code);
+      }
+    );
+  },
+
+  nodeToRange(node) {
+    return node.range;
```

Comment:
if there is infinite recursion issue I guess check if node.range is undefined and return []